DNN Platform/Library/Entities/Urls/AdvancedUrlRewriter.cs

```diff
@@ -1554,8 +1554,11 @@ private static bool CheckForDebug(HttpRequest request, NameValueCollection query
                     //check ssl enforced
                     if (portalSettings.SSLEnforced)
                     {
-                        //check page is not secure, connection is secure
-                        if (!portalSettings.ActiveTab.IsSecure && result.IsSecureConnection)
+                        // Prevent browser's mixed-content error in case we open a secure PopUp or a secure iframe 
+                        // from an unsecure page
+                        if (!portalSettings.ActiveTab.IsSecure && 
+                            result.IsSecureConnection &&
+                            UrlUtils.IsPopUp(url))

```

Comment:
@mean2me I think now the `UrlUtils.IsPopUp(url)` part of the condition got reversed. It should be negated to be equivalent to your initial implementation:
```c#
if (!portalSettings.ActiveTab.IsSecure && result.IsSecureConnection && !UrlUtils.IsPopUp(url))
```

Reply:
@daguiler  I think my initial implementation is the right one. 
Please, have a look at this spreadshit I created: https://docs.google.com/spreadsheets/d/1IPE7QsC1WPiLLlhC4A0j459rw3RN7oTfRh6jv_2YtAQ/edit?usp=sharing
The column **previous** implements a formula reproducing my initial pull request and, actually, it's the expected condition.

Reply:
@daguiler I think current implementation is correct, since it reproduce expected results. If you apply your suggestion, we get a wrong condition.
Please, have a look at this spreadsheet: https://docs.google.com/spreadsheets/d/1IPE7QsC1WPiLLlhC4A0j459rw3RN7oTfRh6jv_2YtAQ/edit?usp=sharing

Reply:
@mean2me Latest results are different from your original results. This is because your latest implementation is not equivalent to your original implementation, or the implementation suggested by me.

Your original implementation was:
```c#
isPopUpFromSecurePage = result.IsSecureConnection && UrlUtils.IsPopUp(url)
isSecureConnectionOnUnsecurePage = !portalSettings.ActiveTab.IsSecure && result.IsSecureConnection

if (isSecureConnectionOnUnsecurePage && !isPopUpFromSecurePage)
{
}
```
Decomposing and simplifying above code, it boils down to:
```c#
if (!portalSettings.ActiveTab.IsSecure && result.IsSecureConnection && !UrlUtils.IsPopUp(url))
{
}
```
Notice the last part is negative, not positive.

Also in your spreadsheet, you are missing the most important combination according to the original implementation: false, true, false.

Here's why I say latest implementation is wrong and my suggestion is equivalent to the original:

![dnn-26576](https://user-images.githubusercontent.com/30123437/50307070-18309580-0476-11e9-91c8-d588431c6120.png)

The important condition (to enter the `if` statement) is highlighted in blue.

**So 2 options now:**

1) Original implementation was correct --> Take my suggestion
2) Original implementation was not correct --> Leave as is

Please confirm. Thanks!

Reply:
@daguiler 
The combination I want to be true is (following above columns' order): FALSE, TRUE, TRUE.
It means: 

- *ActiveTab.IsSecure = FALSE*, so I'm opening a *popup* or an *iframe* from an *unsecure page*.

- *IsSecureConnection = TRUE*: PopUp/iFrame content is secure

- *IsPop = TRUE*: It returns true for both *PupUps and iFrames*.

Reply:
@daguiler you're right, I checked the history of the file and I could see my misunderstanding.
Fixed now.